library/alloc/src/borrow.rs

```diff
@@ -307,7 +306,7 @@ impl<B: ?Sized + ToOwned> Cow<'_, B> {
     /// );
     /// ```
     ///
-    /// Calling `into_owned` on a `Cow::Owned` is a no-op:
+    /// Calling `into_owned` on a `Cow::Owned` returns the owned data:
```

Comment:
I think it's important to explicitly say that it's the original owned data and doesn't involve any new allocations/newly cloned data.

Maybe something like this:

```suggestion
    /// Calling `into_owned` on a `Cow::Owned` returns the owned data unmodified, without cloning it:
```

(I see that the example doesn't really show that. Maybe a better example would show that the pointer (`cow.as_ptr()`) doesn't change before and after `into_owned()`.)

Reply:
I've updated the text to say that the value is moved and not cloned. I did not change the example, as the two examples demonstrate that we get the same outcome regardless of the Cow variant. Adding code to the Owned version to prove that the additional statement is true would obscure this.